File: 189.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
        int n = nums.size();
        vector<int> copy(n);
            
        if ( (n==0) || (k<=0) )
            return;
        
        for(int i=0; i<n; i++)
            copy[i] = nums[i];
        
        //rotating
        
        for(int i=0; i<n; i++){
            nums[(i+k)%n] = copy[i];
        }
    }
};

File: Py/189.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        
        temp = [None] #O(1) extra space
        for i in range(k):
            temp[0] = nums.pop()
            nums.insert(0,temp[0])